code/game/objects/effects/anomalies/anomaly_bleed.dm

```diff
@@ -0,0 +1,60 @@
+/obj/effect/anomaly/blood
+	name = "blood anomaly"
+	icon_state = "blood_anomaly"
+	density = TRUE
+	aSignal = /obj/item/assembly/signaler/anomaly/blood
+	var/sucking = FALSE
+
+/obj/effect/anomaly/blood/anomalyEffect(delta_time)
+	if (sucking)
+		return
+	..()
+	if (DT_PROB(20, delta_time))
+		AddComponent(/datum/component/pellet_cloud, projectile_type=/obj/projectile/bullet/shrapnel/bleed, magnitude=3)
+		playsound(src, 'sound/weapons/shrapnel.ogg', 70, TRUE)
+	if (DT_PROB(10, delta_time))
+		for (var/mob/living/carbon/bleed_target in view(6, src))
+			if (!bleed_target.resting)
+				continue
+			if (do_teleport(src, get_turf(bleed_target), channel = TELEPORT_CHANNEL_BLUESPACE))
+				bleed_target.Stun(3 SECONDS)
+				INVOKE_ASYNC(src, PROC_REF(suck_blood))
+			break
+
+/obj/effect/anomaly/blood/detonate()
```

Comment:
I think this needs a failsafe to prevent detonate in centcom, like `if(!is_station(src.z)) return`, so that people won't get attacked randomly by an admin anomaly.

Reply:
This is kind of fair, but if an admin allows an anomaly to detonate on Centcom then its kind of their fault. Bluespace, Flux, Hallucination, Insanity, and BHole have effects that will affect the station.

Reply:
If this is blocking then I can make the changes, snice they are relatively simple

Reply:
> This is kind of fair, but if an admin allows an anomaly to detonate on Centcom then its kind of their fault. Bluespace, Flux, Hallucination, Insanity, and BHole have effects that will affect the station.

because the result of this proc, detonate(), is only limited to the station because of line 28. It sounds weird when a detonated anomaly from a non-station area like CC or Lavaland only affects to station regardless of the distance.

Reply:
As I mentioned before, most anomalies will affect the station in some way even if spawned on CentCom. Coders shouldn't have to code around admins incorrectly using tools made for debugging. If this is an issue, then its better to make this as a github issue as its a problem with other anomalies and isn't specific to the bleed anomaly.